shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -633,6 +641,13 @@ - (void)goToApplicationLifecycle:(nonnull NSString*)state {
   // Accessing self.view will create the view. Check whether the view is organically loaded
   // first before checking whether the view is attached to window.
   if (self.isViewLoaded && self.view.window) {
+    UIApplicationState appState = [UIApplication sharedApplication].applicationState;
```

Comment:
This logic seems to be reasoning about what happens down below instead of what is the correct behavior at this layer.